.github/workflows/ci.yml

```diff
@@ -59,7 +54,8 @@ jobs:
         run: |
           sudo apt-get update
           sudo apt-get install libkrb5-dev
-          pip install .[tests]
-      - name: Run tests
+          sudo curl -sSL https://install.python-poetry.org | python3 - --preview
+          poetry install --only=dev
+      - name: Run tox
         run: |
-          pytest -s tests/
+          poetry run tox --parallel
```

Comment:
Why is `--parallel` needed?

Reply:
@hashhar rather than multiple GitHub jobs this workflow is configured to have only one job and thus from a wall-clock perspective having the relevant tox environments run in parallel is more desirable.

Reply:
Having single job makes it harder to inspect which env is failing without having to go to the actions page, and check the logs. Any way we can preserve current matrix behaviour?

Reply:
See https://github.com/trinodb/trino-python-client/pull/201#discussion_r931252809. I'm really not certain whether the hypothesized concerns are problematic in reality. There's very little friction in introspecting the GitHub workflow logs to determine why CI is failing.